To-do list/Controllers/TaskDetailsViewController.swift

```diff
@@ -7,39 +7,44 @@
 
 import UIKit
 
+protocol TaskDetailsViewControllerDelegate: AnyObject {
+    func sendTaskDetails(_: TaskDetailsViewController, didCreateUpdate task: Task)
+}
+
 class TaskDetailsViewController: UIViewController {
     
+    weak var delegate: TaskDetailsViewControllerDelegate?
+    
     @IBOutlet weak var titleTextField: UITextField!
     @IBOutlet weak var descriptionTextField: UITextField!
     @IBOutlet weak var statusButtonOutlet: UIButton!
     
-    var users = [User]()
-    var userIndex = Int()
-    var taskIndex = Int()
+    
+
+    var task = Task()
     var counter = 0
-    var status = "to do"
-    var titleField = ""
+    var status = Task.Status.todo
     var choosenTableViewCell = Bool()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        users = UserDefaultsManager.shared.getValueForUser() ?? [User]()
-                titleTextField.text = titleField
+       
         if choosenTableViewCell {
-            titleTextField.text = users[userIndex].task?[taskIndex].title
-            descriptionTextField.text = users[userIndex].task?[taskIndex].description
+            titleTextField.text = task.title
+            descriptionTextField.text = task.description
+            status = task.status!
             
-            switch users[userIndex].task?[taskIndex].status {
-            case "in progress":
+            switch status {
+            case .inProgress:
                 statusButtonOutlet.backgroundColor = .systemOrange
-                statusButtonOutlet.setTitle("in progress", for: .normal)
-            case "done":
+                statusButtonOutlet.setTitle(status.rawValue, for: .normal)
+            case .done:
                 statusButtonOutlet.backgroundColor = .systemRed
-                statusButtonOutlet.setTitle("done", for: .normal)
-            case "to do":
+                statusButtonOutlet.setTitle(status.rawValue, for: .normal)
+            case .todo:
                 statusButtonOutlet.backgroundColor = .systemGreen
-                statusButtonOutlet.setTitle("to do", for: .normal)
+                statusButtonOutlet.setTitle(status.rawValue, for: .normal)
```

Comment:
This all logic should be encapsulated in the TaskButton view. If this a future work - you can leave it for now as is.

Reply:
We will merge this PR and then work on the rest including StatusButton.

Reply:
Status Button class is created.

Reply:
StatusButton should inherit from UIButton, and declared in Storyboard and referenced in controller class.

Reply:
StatusButton class was deleted according to your instructions from 20.09.2022.